src/transformStateWithClones.js

```diff
@@ -65,7 +65,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const result = [];
+  let newState = Object.assign({}, state);
+
+  for (const command of transforms) {
+    switch (command.operation) {
+      case 'addProperties':
+        Object.assign(newState, command.properties);
+        break;
+      case 'removeProperties':
+        for (const key of command.properties) {
+          delete newState[key];
+        }
+        break;
+      case 'clear':
+        newState = {};
+        break;
+      default:
+        return 'You must enter a command';
```

Comment:
```suggestion
       throw new Error('unkown operation');
```
краще завжди повертати з функції дані одного типу (зазвичай у тебе повертається об'єкт а тут повернеться строка), так як важко потім хендлити усі можливі варіанти. Тому тут краще або не робити зовсім нічого, або викидувати помилку, якщо ми впевнені що всі операції будуть правильні і хочемо зразу побачити помилку у випадку якщо хтось спробує преедати неіснуючу операцію